### PR TITLE
fix: return correct relative path if preexisting file ext

### LINKED ISSUE #1064: Reletive paths to static resouces with file extensions get .html appended to them
When using the [`path` handlebars helper](https://fractal.build/guide/core-concepts/view-templates.html#path) to generate relative paths for assets all of the assets are appended with `.html`, even though they have an extension already.

E.g. `pathTo/image.jpg` is being output as `pathTo/image.jpg.html`.

I did some digging into the code and found out this is happening in the `relUrlPath` function within the `core/src/utils.js` file.  Some sluthing on GitHub led me to #575.  This change doesn't verify there are existing extensions and adds the `.html` extension irregardless of the original path name.

It looks like one option is to [override this the default extension](https://fractal.build/guide/web/configuration-reference.html#builder-urls-ext) by adding `fractal.web.set('builder.urls.ext', null);` to one's `fractal.js` configuration file. 

Note: [Relevant message on Discord](https://discord.com/channels/664450585168314409/664450890660446229/864603137474166815).

### Steps to reproduce the issue

1. In a component markdown file use the `path` helper in a handlebars template.
    ```
    {{path '../.../markdown-assets/documentation/image.jpg'}}
    ```
2. Inspect the rendered output HTML, the output is `../.../markdown-assets/documentation/image.jpg.html`

**Reproduces how often:**

`100%`

### Context

-   Fractal version: 1.5.8
-   Node version: v14.17.1
-   OS: Linux Mint 20.2 (Ubuntu 20.04), macOS 11.4

### PR DESCRIPTION
**Pull request by ScottMorris**
This PR closes #1064.

It addresses the issue by using regex to find a file extension on the `toPath` when the path is found to start with a `.`.  The solution also incorporates a test to handle this edge case.

### CODE REVIEW

File: packages/core/src/utils.js [hunk 1]
```diff
@@ -140,7 +140,8 @@ module.exports = {
         const ext = opts.ext || '';
 
         if (toPath.startsWith('.')) {
-            return ext === '' ? toPath : toPath + ext;
+            const toPathExt = /^.+\.([^.]+)$/.exec(toPath);
```
Comment (mihkeleidast) [inline packages/core/src/utils.js:143]: Use `Path.extname(toPath)` as is used a few lines below - `Path.extname(toPath) ? toPath : toPath + ext`.

**Review by mihkeleidast — CHANGES_REQUESTED** [review]

File: packages/core/src/utils.js (hunk 1 above)
Comment (ScottMorris) [reply packages/core/src/utils.js:143]: D'oh, good suggestion!

Comment (ScottMorris) [reply packages/core/src/utils.js]: I have mode this update.

**Review by mihkeleidast — APPROVED** [review]

**Comment by mihkeleidast** [comment]
Very nice, thank you!
